Purchases/Public/RCPurchases.m

```diff
@@ -845,6 +849,7 @@ - (void)collectDeviceIdentifiers {
 - (void)applicationDidBecomeActive:(__unused NSNotification *)notif {
     [self updateAllCachesIfNeeded];
     [self syncSubscriberAttributesIfNeeded];
+    [self postAppleSearchAddsAttributionCollectionIfNeeded];
```

Comment:
calling it on `didBecomeActive`, so it gets picked up right after the prompt, and also if you change the authorization from iOS settings

Reply:
Will this still post if attribution was already sent earlier without the ATT permission accepted? It looks like as long as the any of the values are different than last time it should? Or is `latestNetworkAndAdvertisingIdsSentForAppUserID` just keyed off the network?

Reply:
If the method gets called but authorization wasn't granted, `latestNetworkAndAdvertisingIdsSentForAppUserID` won't be set, because we check for authorization first. So when you call again but with authorization granted, it'll post correctly because the value was never set. 

However, if you are on 14.4, for example, get a value and post, then upgrade to 14.5 and are unauthorized, the value won't be posted again once you're authorized because a value has already been sent.

i.e.: it'll only be sent once, and only if you're authorized. If you're not authorized, it'll keep trying to see if it can send it. If the search ads value changes, though, it won't post again. This was the behavior before this PR, and I figured it made sense since it's not expected to change, right? [Apple's docs](https://developer.apple.com/documentation/iad/setting_up_apple_search_ads_attribution?language=objc) even suggest that you store the values locally so that you don't repeat the call.